src/packages/solid/components/Keyboard/KeyboardSimple.tsx

```diff
@@ -15,61 +15,110 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { For, type Component } from 'solid-js';
-import { type KeyboardProps } from './Keyboard.jsx';
+import { For, Show, createMemo, createSignal, type Accessor, type Component } from 'solid-js';
 import Column from '../Column/Column.jsx';
 import Row from '../Row/Row.jsx';
-import Key, { type KeyProps } from '../Key/Key.jsx';
+import Key from '../Key/Key.jsx';
+import type { KeyProps } from '../Key/Key.types.js';
 import styles from './Keyboard.styles.js';
+import type { KeyboardProps } from './Keyboard.types.js';
+import { ElementNode, View } from '@lightningjs/solid';
+
+const getTone = (props: KeyProps) => props.tone ?? styles.tone;
 
 // rows created from each array passed in
 const KeyboardSimple: Component<KeyboardProps> = (props: KeyboardProps) => {
   // eslint-disable-next-line solid/reactivity
-  const [_, setKeySignal] = props.keySignal;
+  const [_, setKeySignal] = props.keySignal ?? createSignal('');
+  const [activeKeyboard, setActiveKeyboard] = createSignal('default');
+  const [selectedRowIndex, setSelectedRowIndex] = createSignal(0);
+  const [selectedColumnIndex, setSelectedColumnIndex] = createSignal(0);
+
+  const setOnEnter = (key: string | KeyProps, rowIdx: Accessor<number>, colIdx: Accessor<number>) => {
+    if (typeof key === 'string') {
+      return () => setKeySignal(key);
+    } else if (key.toggle) {
+      return () => {
+        setSelectedRowIndex(rowIdx());
+        setSelectedColumnIndex(colIdx());
+        setActiveKeyboard(key.toggle);
+        keyboardRef[key.toggle]?.setFocus();
+      };
+    } else {
+      return () => setKeySignal(typeof key === 'string' ? key : key.title ?? '');
+    }
```

Comment:
is it possible to do this without `typeof` checks? I'm aware we were using them before but it's a pattern we should avoid if we can

Reply:
I cant think of a better way to check if its a string vs an object. I can keep looking but passing an object into the signal will break